src/lib/APS2Ethernet.cpp

```diff
@@ -0,0 +1,560 @@
+// Ethernet communications with APS2 boards
+//
+// Original authors: Colm Ryan, Blake Johnson, Brian Donovan
+// Copyright 2016, Raytheon BBN Technologies
+
+#include "APS2Ethernet.h"
+
+#ifdef _WIN32
+#include "iphlpapi.h"
+#else
+#include <ifaddrs.h>
+#endif
+
+APS2Ethernet::APS2Ethernet() : udp_socket_old_(ios_), udp_socket_(ios_) {
+    FILE_LOG(logDEBUG) << "APS2Ethernet::APS2Ethernet";
+
+    //Move the io_service to a background thread
+    //Give it something to chew on so that it doesn't return from the background thread
+    asio::io_service::work work(ios_);
+    receiveThread_ = std::thread([this](){ ios_.run(); });
+
+    //Bind the old UDP socket at local port bb4e
+    try {
+        udp_socket_old_.open(udp::v4());
+        udp_socket_old_.bind(udp::endpoint(udp::v4(), APS_PROTO));
+    } catch (...) {
+        throw APS2_SOCKET_FAILURE;
+    }
+
+    //Bind UDP socket at local port bb4f
+    try {
+        udp_socket_.open(udp::v4());
+        udp_socket_.bind(udp::endpoint(udp::v4(), APS_PROTO+1));
+    } catch (...) {
+        throw APS2_SOCKET_FAILURE;
+    }
+
+    //enable broadcasting for enumerating
+    udp_socket_old_.set_option(asio::socket_base::broadcast(true));
+    udp_socket_.set_option(asio::socket_base::broadcast(true));
+
+    setup_udp_receive_old();
+    setup_udp_receive();
+
+};
+
+APS2Ethernet::~APS2Ethernet() {
+    FILE_LOG(logDEBUG) << "Cleaning up ethernet interface";
+    ios_.stop();
+    receiveThread_.join();
+}
+
+void APS2Ethernet::setup_udp_receive_old() {
+    //Receive UDP packets on the old UDP sockets
+    udp_socket_old_.async_receive_from(
+        asio::buffer(received_udp_data_old_, 2048), remote_udp_endpoint_old_,
+        [this](std::error_code ec, std::size_t bytesReceived)
+        {
+            //If there is anything to look at hand it off to the sorter
+            if (!ec && bytesReceived > 0)
+            {
+                vector<uint8_t> packetData(received_udp_data_old_, received_udp_data_old_ + bytesReceived);
+                sorter_lock_.lock();
+                sort_packet(packetData, remote_udp_endpoint_old_);
+                sorter_lock_.unlock();
+            }
+
+            //Start the receiver again
+            setup_udp_receive_old();
+    });
+}
```

Comment:
These two methods are the same except of the values `received_udp_data_old_` and `remote_udp_endpoint_old_`, no? So, can we make those arguments to the function, instead?